src/App.tsx

```diff
@@ -1,12 +1,57 @@
+import { useState } from 'react';
+import './App.css';
+
+// components
 import { AppHeader } from './components/AppHeader';
 import { AppLayout } from './components/AppLayout';
+import NewCard from './components/Card/NewCard';
+import CardsList from './components/Card/CardsList';
 
-import './App.css';
+interface CardData {
+  firstPage: string;
+  secondPage: string;
+}
 
 function App() {
+  const [cards, setCards] = useState<
+    { firstPage: string; secondPage: string }[]
+  >([]);
+
+  const [newCardIsAdded, setNewCardIsAdded] = useState(false);
+
+  const addNewCard = () => {
+    setNewCardIsAdded(true);
+  };
+
+  const cancelNewCard = () => {
+    setNewCardIsAdded(false);
+  };
+
+  const saveNewCard = (cardData: CardData) => {
+    setCards((prevCards) => [
+      {
+        firstPage: cardData.firstPage,
+        secondPage: cardData.secondPage,
+      },
+      ...prevCards,
+    ]);
+    setNewCardIsAdded(false);
+  };
+
   return (
     <AppLayout>
-      <AppHeader cardsAmount={0} />
+      <AppHeader onClick={addNewCard} cardsAmount={cards.length} />
+
+      {newCardIsAdded && (
+        <NewCard onSaveNewCard={saveNewCard} onCancelNewCard={cancelNewCard} />
+      )}
+
+      {cards && <CardsList cards={cards} />}
+      {cards.length === 0 && !newCardIsAdded && (
+        <p style={{ color: 'var(--typo-color)', margin: '27px 0 0 16px' }}>
```

Comment:
**nitpick:** Writing inline styles is not a good practice, as styles object has to be recreated on every render. It can lead to worse performance when it's overused. You can solve it by putting this styles object outside component body